pkg/dogstatsd/replay/file.go

```diff
@@ -68,15 +69,15 @@ func fileVersion(buf []byte) (int, error) {
 }
 
 // WriteHeader writes the datadog header to the Writer argument to conform to the .dog file format.
-func WriteHeader(w *bufio.Writer) error {
+func WriteHeader(w io.Writer) error {
 	hdr := make([]byte, len(datadogHeader))
 	copy(hdr, datadogHeader)
 	hdr[versionIndex] |= datadogFileVersion
 
 	//Write header
 	if n, err := w.Write(hdr); err != nil || n < len(datadogHeader) {
```

Comment:
Feels like this might be clearer rewritten into two `if` statements, wdyt?

```
n, err := w.Write(hdr)
if err != nil {
    return err
}
if n < len(datadogHeader) {
    return ErrHeaderWrite
}
```

(it hurts my brain a bit to have two conditions but then re-testing for the _inverse_ of one of those conditions)

Reply:
Yep, that makes sense to me 👍🏽